backend/v3/domain/org_update.go

```diff
@@ -0,0 +1,66 @@
+package domain
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/backend/v3/storage/database"
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type UpdateOrgCommand struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+var _ Commander = (*UpdateOrgCommand)(nil)
+
+func (u *UpdateOrgCommand) Execute(ctx context.Context, opts *CommandOpts) (err error) {
+	close, err := opts.EnsureTx(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() { err = close(ctx, err) }()
+
+	organizationRepo := orgRepo(opts.DB)
+
+	updateCount, err := organizationRepo.Update(
+		ctx,
+		organizationRepo.IDCondition(u.ID),
+		authz.GetInstance(ctx).InstanceID(),
+		database.NewChange(organizationRepo.NameColumn(), u.Name),
+	)
+	if err != nil {
+		return err
+	}
+
+	if updateCount == 0 {
+		err = zerrors.ThrowNotFound(nil, "DOM-7PfSUn", "organization not found")
+		return err
+	}
+	if updateCount > 1 {
+		err = zerrors.ThrowInternalf(nil, "DOM-QzITrx", "expecting 1 row updated, got %d", updateCount)
+		return err
+	}
+
+	orgCache.Set(ctx, &Organization{
```

Comment:
caching is currently in discussion, so I would not use it for the moment because we might adjust the interface and would break the code